react-playground/src/components/Header/Header.tsx

```diff
@@ -1,51 +1,47 @@
-import { FC, useState } from "react";
+import { FC, useState, useLayoutEffect, useRef, useCallback } from "react";
 import { Link } from "react-router-dom";
 import MainLogo from "../../images/main-logo.svg";
 import MenuIcon from "../../images/menu-icon.svg";
-import { HeaderStyled, DropdownStyled, MenuItemStyled } from "./Header.styles";
-
-const menu = [
-  { name: "Home", route: "/" },
-  { name: "My Results", route: "/results" },
-  { name: "Health profile", route: "/healthprofile" },
-  { name: "Community", route: "/community" },
-  { name: "Resources", route: "/resources" },
-  { name: "Settings", route: "/settings" },
-  { name: "Log out", route: "/logout" },
-];
+import { HeaderStyled, MenuIconStyled } from "./Header.styles";
+import { Menu } from "../Menu";
 
 const Header: FC = () => {
   const [isOpen, setIsOpen] = useState(false);
   const toggling = () => setIsOpen(!isOpen);
 
-  const onItemClicked = (value: Object) => () => {
+  const onItemClicked = () => {
     setIsOpen(false);
   };
 
+  const element = useRef<HTMLDivElement>(null);
+
+  const handleClick = useCallback((e: MouseEvent) => {
+    if (element.current && element.current.contains(e.target as HTMLElement)) {
+      return;
+    }
+    setIsOpen(false);
+  }, []);
+
+  useLayoutEffect(() => {
```

Comment:
I'd avoid using useLayoutEffect unless absolutely necessary – In this particular context, because what's inside of useLayoutEffect does not involve anything that needs to change immediately before the components are drawn to the screen, we shouldn't need it. I think it can theoretically slow things down, so just use useEffect here. https://dev.to/nibble/what-is-uselayouteffect-hook-and-when-do-you-use-it-3lan